File: backend/app/schemas/publish.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PublishResponse(BaseModel):
    publish_run_id: int
    outcome: str
    shows_count: int
    episodes_count: int
    catalogue_uri: str | None = None
    error_message: str | None = None


class PublishRunResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    published_by: int
    started_at: datetime
    completed_at: datetime | None
    shows_count: int
    episodes_count: int
    outcome: str
    catalogue_uri: str | None
    error_message: str | None
    created_at: datetime

File: frontend/src/api/publish.ts
import api from "./client";

export interface PublishResponse {
  publish_run_id: number;
  outcome: string;
  shows_count: number;
  episodes_count: number;
  catalogue_uri?: string | null;
  error_message?: string | null;
}

export interface PublishRun {
  id: number;
  published_by: number;
  started_at: string;
  completed_at: string | null;
  shows_count: number;
  episodes_count: number;
  outcome: string;
  catalogue_uri: string | null;
  error_message: string | null;
  created_at: string;
}

export async function publishCatalogue(): Promise<PublishResponse> {
  const response = await api.post<PublishResponse>("/publish");
  return response.data;
}

export async function getPublishRuns(): Promise<PublishRun[]> {
  const response = await api.get<PublishRun[]>("/publish/runs");
  return response.data;
}

export async function getPublishRun(
  id: number
): Promise<PublishRun> {
  const response = await api.get<PublishRun>(
    `/publish/runs/${id}`
  );
  return response.data;
}
